### ISSUE TITLE
E121: Undefined variable: subcommands E15: Invalid expression: meta.subcommands

### ISSUE THREAD

**Issue by lkraav** [dsawardekar/wordpress.vim#2]
Today's master gives the above errors on vim launch.

```
functions.php                                                                                                                                                                                                                55,1           Top
WordPress: Loading ...
Error detected while processing function <SNR>43_s:WordPressPlugin_on_buffer_enter..<SNR>83_s:App_on_buffer_enter..<SNR>83_s:Controller_process..<SNR>83_s:CommandRegistry_run_action..<SNR>83_s:CommandRegistry_run_command..<SNR>83_s:ChangeC
urrentBufferCommand_run..<SNR>83_s:ProjectCollection_add..<SNR>83_s:Project_load..<SNR>83_s:WpCli_list:
line    3:
E121: Undefined variable: subcommands
E15: Invalid expression: meta.subcommands
```

```
$ [git:master+?] vim --version
VIM - Vi IMproved 7.3 (2010 Aug 15, compiled Jul  4 2013 00:12:40)
Included patches: 1-1214
Modified by Gentoo-7.3.1214
Huge version without GUI.  Features included (+) or not (-):
+arabic          +file_in_path    +mouse_sgr       +tag_binary
+autocmd         +find_in_path    -mouse_sysmouse  +tag_old_static
-balloon_eval    +float           +mouse_urxvt     -tag_any_white
-browse          +folding         +mouse_xterm     -tcl
++builtin_terms  -footer          +multi_byte      +terminfo
+byte_offset     +fork()          +multi_lang      +termresponse
+cindent         +gettext         -mzscheme        +textobjects
-clientserver    -hangul_input    +netbeans_intg   +title
-clipboard       +iconv           +path_extra      -toolbar
+cmdline_compl   +insert_expand   -perl            +user_commands
+cmdline_hist    +jumplist        +persistent_undo +vertsplit
+cmdline_info    +keymap          +postscript      +virtualedit
+comments        +langmap         +printer         +visual
+conceal         +libcall         +profile         +visualextra
+cryptv          +linebreak       -python          +viminfo
-cscope          +lispindent      -python3         +vreplace
+cursorbind      +listcmds        +quickfix        +wildignore
+cursorshape     +localmap        +reltime         +wildmenu
+dialog_con      -lua             +rightleft       +windows
+diff            +menu            -ruby            +writebackup
+digraphs        +mksession       +scrollbind      -X11
-dnd             +modify_fname    +signs           -xfontset
-ebcdic          +mouse           +smartindent     -xim
+emacs_tags      -mouseshape      -sniff           -xsmp
+eval            +mouse_dec       +startuptime     -xterm_clipboard
+ex_extra        -mouse_gpm       +statusline      -xterm_save
+extra_search    -mouse_jsbterm   -sun_workshop    
+farsi           +mouse_netterm   +syntax          
   system vimrc file: "/home/markitek/prefix/etc/vim/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  fall-back for $VIM: "/home/markitek/prefix/usr/share/vim"
Compilation: x86_64-pc-linux-gnu-gcc -c -I. -Iproto -DHAVE_CONFIG_H     -O2 -pipe -O2 -pipe -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1      
Linking: x86_64-pc-linux-gnu-gcc   -Wl,-O1 -Wl,--as-needed -o vim        -lm -lncurses -lelf -lnsl  -ldl          
```

**Comment by dsawardekar** [comment]
That's odd, it's detecting the presence of `wp-cli` but not getting a `subcommands` array. Can you post the output of `wp cli cmd-dump`? It should return JSON that the plugin loops over.

Edit:

With this bug you won't be able to run any of the `W..` ex commands. Can you post the direct output of `wp cli info`. Thanks.

**Comment by dsawardekar** [comment]
Can you please update to `0.1.7`. I've included more robust checks for `WpCli#list` method. If you are still facing issues please post the direct wp-cli details from above.

**Comment by dsawardekar** [comment]
I'm considering this as fixed after the changes in `0.1.7`. Please reopen with above details if you are still having problems.
